File: iOS/Code/Xamarin/bugTrap/bugTrapKit/Trackers/JIRA/JiraDetail.cs
namespace bugTrapKit.Jira
{
	public enum JiraDetail
	{
		Images,
		Project,
		IssueType,
		Summary,
		Priority,
		Description,
		Assignee,
		DueDate,
		Environment,
		FixVersions,
		Versions,
		Labels
	}

	public static class JiraDetailExtensions
	{
		public static string SettingsKey (this JiraDetail detail)
		{
			switch (detail) {
			case JiraDetail.Images:
				return "JiraDetailImages";
			case JiraDetail.Project:
				return "JiraDetailProject";
			case JiraDetail.IssueType:
				return "JiraDetailIssueType";
			case JiraDetail.Summary:
				return "JiraDetailSummary";
			case JiraDetail.Priority:
				return "JiraDetailPriority";
			case JiraDetail.Description:
				return "JiraDetailDescription";
			case JiraDetail.Assignee:
				return "JiraDetailAssignee";
			case JiraDetail.DueDate:
				return "JiraDetailDueDate";
			case JiraDetail.Environment:
				return "JiraDetailEnvironment";
			case JiraDetail.FixVersions:
				return "JiraDetailFixVersions";
			case JiraDetail.Versions:
				return "JiraDetailVersions";
			case JiraDetail.Labels:
				return "JiraDetailLabels";
			default:
				return null;
			}
		}

		public static bool IsPersonDetail (this JiraDetail detail)
		{
			switch (detail) {
			case JiraDetail.Assignee:
				return true;
			default:
				return false;
			}
		}
	}
}

File: iOS/Code/Swift/bugTrap/bugTrapKit/Trackers/JIRA/JiraDetail.swift
//
//  JiraDetail.swift
//  bugTrap
//
//  Created by Colby L Williams on 10/19/14.
//  Copyright (c) 2014 bugTrap. All rights reserved.
//

import Foundation

enum JiraDetail: Int, TrackerDetailType {
	case Images = 0, Project, IssueType, Summary, Priority, Description, Assignee, DueDate, Environment, FixVersions, Versions, Labels

	func isPersonDetail () -> Bool {
        switch self {
		case .Assignee: return true
		default: return false
        }
    }

	var settingsKey: String {
		switch self {
			case .Images: return "JiraDetailImages"
			case .Project: return "JiraDetailProject"
			case .IssueType: return "JiraDetailIssueType"
			case .Summary: return "JiraDetailSummary"
			case .Priority: return "JiraDetailPriority"
			case .Description: return "JiraDetailDescription"
			case .Assignee: return "JiraDetailAssignee"
			case .DueDate: return "JiraDetailDueDate"
			case .Environment: return "JiraDetailEnvironment"
			case .FixVersions: return "JiraDetailFixVersions"
			case .Versions: return "JiraDetailVersions"
			case .Labels: return "JiraDetailLabels"
		}
	}
}